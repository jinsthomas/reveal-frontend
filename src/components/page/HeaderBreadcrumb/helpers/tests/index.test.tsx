import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { createBrowserHistory } from 'history';
import React from 'react';
import { Router } from 'react-router';
import { LinkList } from '..';

const history = createBrowserHistory();

describe('components/page/HeaderBreadcrumb/helpers', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders without crashing', () => {
    const props = {
      pages: [
        {
          label: 'Home',
          url: '/',
        },
      ],
    };
    shallow(
      <Router history={history}>
        <LinkList {...props} />
      </Router>
    );
  });

  it('renders LinkList correctly', () => {
    const props = {
      pages: [
        {
          label: 'Home',
          url: '/',
        },
        {
          label: 'Programs',
          url: '/programs',
        },
        {
          label: 'Provinces',
          url: '',
        },
        {
          label: 'Disctricts',
        },
      ],
    };
    const wrapper = mount(
      <Router history={history}>
        <LinkList {...props} />
      </Router>
    );
    expect(toJson(wrapper.find('LinkList'))).toMatchSnapshot();
    wrapper.unmount();
  });
});
