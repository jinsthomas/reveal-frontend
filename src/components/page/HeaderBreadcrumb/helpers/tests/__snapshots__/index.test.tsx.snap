// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/page/HeaderBreadcrumb/helpers renders LinkList correctly 1`] = `
<LinkList
  pages={
    Array [
      Object {
        "label": "Home",
        "url": "/",
      },
      Object {
        "label": "Programs",
        "url": "/programs",
      },
      Object {
        "label": "Provinces",
        "url": "",
      },
      Object {
        "label": "Disctricts",
      },
    ]
  }
>
  <BreadcrumbItem
    key="0"
    tag="li"
  >
    <li
      className="breadcrumb-item"
    >
      <Link
        key="0"
        to="/"
      >
        <LinkAnchor
          href="/"
          navigate={[Function]}
        >
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </LinkAnchor>
      </Link>
    </li>
  </BreadcrumbItem>
  <BreadcrumbItem
    key="1"
    tag="li"
  >
    <li
      className="breadcrumb-item"
    >
      <Link
        key="1"
        to="/programs"
      >
        <LinkAnchor
          href="/programs"
          navigate={[Function]}
        >
          <a
            href="/programs"
            onClick={[Function]}
          >
            Programs
          </a>
        </LinkAnchor>
      </Link>
    </li>
  </BreadcrumbItem>
  <BreadcrumbItem
    key="2"
    tag="li"
  >
    <li
      className="breadcrumb-item"
    >
      Provinces
    </li>
  </BreadcrumbItem>
  <BreadcrumbItem
    key="3"
    tag="li"
  >
    <li
      className="breadcrumb-item"
    >
      Disctricts
    </li>
  </BreadcrumbItem>
</LinkList>
`;
